api/types/container/host_config.go

```diff
@@ -284,6 +284,17 @@ type DeviceMapping struct {
 	CgroupPermissions string
 }
 
+// HugepageLimit corresponds to the file`hugetlb.<hugepagesize>.limit_in_bytes` in container level cgroup.
+// For example, `PageSize=1GB`, `Limit=1073741824` means setting `1073741824` bytes to hugetlb.1GB.limit_in_bytes.
+type HugepageLimit struct {
+	// The value of PageSize has the format <size><unit-prefix>B (2MB, 1GB),
+	// and must match the <hugepagesize> of the corresponding control file found in `hugetlb.<hugepagesize>.limit_in_bytes`.
+	// The values of <unit-prefix> are intended to be parsed using base 1024("1KB" = 1024, "1MB" = 1048576, etc).
+	PageSize string
+	// limit in bytes of hugepagesize HugeTLB usage.
+	Limit int64
```

Comment:
Because of this https://github.com/moby/moby/pull/40160#discussion_r360065702,
@cpuguy83 said `It's more in line with (most) of the rest of the API.`
Would it be better to use uint64?

Reply:
Ah, sorry, missed that comment. I guess at least we should look if the API (swagger) should be updated to match that, but also check/validate values.

@cpuguy83 any thoughts?

Reply:
the runtime will fail values that are stupid, we should not check here.